pkg/actuators/machine/actuator.go

```diff
@@ -115,6 +115,9 @@ func (a *Actuator) Create(context context.Context, cluster *machinev1.Cluster, m
 		}
 		return err
 	}
+	if err := a.updateProviderID(machine, instance); err != nil {
```

Comment:
This needs to return machine so that the following call is using the same object.

Reply:
`a.updateProviderID` updates the machine object as well (though not perfect).